nursery/reverse.go

```diff
@@ -222,3 +225,126 @@ func (nursery *Nursery) handleReverseSwapStatus(reverseSwap *database.ReverseSwa
 
 	nursery.sendReverseSwapUpdate(*reverseSwap)
 }
+
+func (nursery *Nursery) queryExternalPendingSwaps(currency boltz.Currency) ([]*database.ReverseSwap, error) {
+	reverseSwaps, err := nursery.database.QueryReverseSwaps(database.SwapQuery{
+		To:     &currency,
+		States: []boltzrpc.SwapState{boltzrpc.SwapState_PENDING, boltzrpc.SwapState_ERROR},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return slices.DeleteFunc(reverseSwaps, func(swap *database.ReverseSwap) bool {
+		return !swap.ExternalPay
+	}), nil
+}
+
+func (nursery *Nursery) searchClaimTx(notification wallet.TransactionNotification) error {
```

Comment:
not sure how to name this

Reply:
I don't like the claim transaction in the name because it is for what you called "direct payments"